### PR TITLE
dev-libs/pigpio: new version 74

### PR DESCRIPTION
**Pull request by github-k8n**
new upstream version, adjust download URL

Signed-off-by: Daniel Kenzelmann <gentoo@k8n.de>

### CODE REVIEW

**Comment by github-k8n** [comment]
Add small change to use package provided systemd unit file.

File: dev-libs/pigpio/pigpio-74.ebuild [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
+
+PYTHON_COMPAT=( python3_{6,7,8} )
+
+inherit distutils-r1 systemd toolchain-funcs
+
+DESCRIPTION="A library for the Raspberry which allows control of the GPIOs"
+HOMEPAGE="http://abyz.me.uk/rpi/pigpio/index.html"
```
Comment (juippis) [inline dev-libs/pigpio/pigpio-74.ebuild]: Please also add the Github page here as a secondary homepage.

File: dev-libs/pigpio/pigpio-74.ebuild [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+LICENSE="Unlicense"
+SLOT="0"
+KEYWORDS="~arm"
+IUSE="python"
+REQUIRED_USE="python? ( ${PYTHON_REQUIRED_USE} )"
+
+src_prepare() {
+	eapply "${FILESDIR}/${PN}-70-makefile.patch"
+	eapply_user
+}
```
Comment (juippis) [inline dev-libs/pigpio/pigpio-74.ebuild]: You can just use a global `PATCHES=( ... )` array here for the patch file, then you can remove these 4 lines.

File: dev-libs/pigpio/pigpio-74.ebuild [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+
+src_install() {
+	emake DESTDIR="${D}" LDCONFIG=: PYTHON2=: PYTHON3=: \
+		libdir="${EPREFIX}/usr/$(get_libdir)" prefix="${EPREFIX}/usr" \
+		mandir="${EPREFIX}/usr/share/man" install
+	einstalldocs
+	newinitd "${FILESDIR}"/pigpiod.initd pigpiod
+	newconfd "${FILESDIR}"/pigpiod.confd pigpiod
+	systemd_newunit "${WORKDIR}"/"${P}"/util/pigpiod.service pigpiod.service
```
Comment (juippis) [inline dev-libs/pigpio/pigpio-74.ebuild]: `"${WORKDIR}"/"${P}"` should be the current working directory, so just calling `systemd_newunit util/pigpiod.service pigpiod.service` should work and be a bit more elegant. Or `${S}`.

**Comment by github-k8n** [comment]
All updated and rebased. Thanks for the tips!

**Pull request closed without merging**